### ISSUE TITLE
Add support for Metal VRF on Metal Billed Interconnections

### ISSUE THREAD

**Issue by displague** [equinix/terraform-provider-equinix#360]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Description

The following new (beta) capabilities should be surfaced in the provider:

- VRF on Metal Billed Fabric VCs
- Setting of VRF Static Route
- Support for BGP Dynamic Neighbors (available on VRF Metal Gateways)
- (non-VRF) Adding a static/elastic IP address to a Gateway

Identified changes (there may be others):
* equinix_metal_connection: should support vrfs as a mutually exclusive option to vlans
* [TODO]

Packngo will not be getting these new capabilities so metal-go (#353) (which already includes support for these API endpoints, parameters, and attributes) will need to be used.

### New or Affected Resource(s)

The following resources and datasources should be updated (including their documentation and expamples) to reflect the new capabilities:

•	Connection (Example Use-Cases): https://registry.terraform.io/providers/equinix/equinix/latest/docs/resources/equinix_metal_connection
•	Virtual Circuit: https://registry.terraform.io/providers/equinix/equinix/latest/docs/resources/equinix_metal_virtual_circuit
•	VRF (Example Use-Cases):
https://registry.terraform.io/providers/equinix/equinix/latest/docs/resources/equinix_metal_vrf
•	IP Reservation (VRF Example Use-Cases):
https://registry.terraform.io/providers/equinix/equinix/latest/docs/resources/equinix_metal_reserved_ip_block#type
•	Metal + Fabric Guide: https://registry.terraform.io/providers/equinix/equinix/latest/docs/guides/equinix_fabric_cloud_providers
•	Metal VRF Module: https://github.com/equinix-labs/terraform-equinix-metal-vrf
 

### Potential Terraform Configuration

```tf
resource "equinix_metal_connection" "example" {
    name            = "tf-port-to-metal-legacy"
    project_id      = local.my_project_id
    metro           = "SV"
    redundancy      = "redundant"
    type            = "shared"
    contact_email   = "username@example.com"

    # Similar to the VLANs feature which is for Shared Interconnection only, expose a VRF parameter
    # vlans = []
    vrf_id = equinix_metal_vrf.example.id
}

resource "equinix_metal_vrf" "example" { ... }
```

### References

* https://deploy.equinix.com/developers/docs/metal/interconnections/vrf/
* https://feedback.equinixmetal.com/network/p/virtual-routing-forwarding-vrf-for-overlapping-ip-space-on-dedicated-and-shared
* https://feedback.equinixmetal.com/network/p/bgp-dynamic-neighbor-feature-parity-for-metal-gateways
* https://feedback.equinixmetal.com/network/p/metal-gateway-elastic-ip
* https://github.com/equinix/terraform-provider-equinix/issues/309 may have relevant changes identified

**Comment by DailyAlice** [comment]
Docs link for adding an Elastic (Reserved) IP address to a Metal Gateway - https://deploy.equinix.com/developers/docs/metal/layer2-networking/metal-gateway/#adding-an-ip-address-to-a-metal-gateway

**Comment by ctreatma** [comment]
Closed by #607
